xwiki-platform-core/xwiki-platform-mail/xwiki-platform-mail-send/xwiki-platform-mail-send-default/src/main/java/org/xwiki/mail/script/ScriptMimeMessage.java

```diff
@@ -129,6 +129,28 @@ public BodyPart addPart(String mimeType, Object content, Map<String, Object> par
         return bodyPart;
     }
 
+    /**
+     * Add some content to the mail to be sent. Can be called several times to add different content type to the mail.
+     * Although BodyParts cannot be created in Velocity, this method is useful when using Groovy
+     * @return the Mime Body Part object that was added. Returning it allows script to make modifications to that body
+     *         part after it's been set (get/set some headers, etc)
+     * @param bodyPart the Mime Body Part object to be added
+     */
+    public BodyPart addPart(BodyPart bodyPart)
```

Comment:
There is not reason to add a new method. You should check if the object is already a BodyPart in the other method instead.

Reply:
@tmortagne How do you check if it is a BodyPart using just the other parameters? Isn't a BodyPart parameter necessary?

Reply:
The other method takes an Object and Object can be anything. You can test the type of the input using `instanceof` in Java.

Reply:
@tmortagne Oh sorry, I did not notice the other method 👍